bookshelf/storage.py

```diff
@@ -0,0 +1,73 @@
+# Copyright 2015 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import absolute_import
+
+import datetime
+import os
+
+from flask import current_app
+from google.cloud import storage
+import six
+from werkzeug import secure_filename
+from werkzeug.exceptions import BadRequest
+
+
+def _check_extension(filename, allowed_extensions):
+    if ('.' not in filename or
```

Comment:
Is there a cleaner way to do this? Maybe [something like this](https://stackoverflow.com/questions/541390/extracting-extension-from-filename-in-python)?

Reply:
Good call! Done.